src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java

```diff
@@ -1013,7 +1022,7 @@ private static Configuration dataFileConfiguration(Project project, String name,
         };
 
         elementsConfiguration.getAllArtifacts().configureEach(artifact -> configurePublishing.run());
-        elementsConfiguration.getArtifacts().configureEach(artifact -> configurePublishing.run());
+        elementsConfiguration.getAllDependencies().configureEach(dep -> configurePublishing.run());
```

Comment:
![image](https://github.com/user-attachments/assets/78be0235-7c4d-4672-84e1-ef13e408173c)

Looks like this is always called because of `dependencies.add(depFactory.create(collection.getPublished()))` above. That counts as a dependency, and there is no resolution that can find out that it's empty.